quantum/joystick.h

```diff
@@ -8,9 +12,18 @@
 #    define JOYSTICK_AXES_COUNT 4
 #endif
 
-#include "quantum.h"
+#ifndef JOYSTICK_AXES_RESOLUTION
+#    define JOYSTICK_AXES_RESOLUTION 8
+#elif JOYSTICK_AXES_RESOLUTION < 8 || JOYSTICK_AXES_RESOLUTION > 16
+#    error JOYSTICK_AXES_RESOLUTION must be between 8 and 16
+#endif
+
+#if JOYSTICK_AXES_RESOLUTION == 16
+#   define JOYSTICK_RESOLUTION 32767
+#else
+#   define JOYSTICK_RESOLUTION (1 << (JOYSTICK_AXES_RESOLUTION - 1)) - 1
+#endif
```

Comment:
```suggestion
#define JOYSTICK_RESOLUTION (1 << (JOYSTICK_AXES_RESOLUTION - 1)) - 1
```
We don't need to do this.

Reply:
I tried to leave this out, but I get integer overflow errors when JOYSTICK_AXES_RESOLUTION is defined as 16

Reply:
Updated, try now.

Reply:
Weird, it compiles, but when I test it, it doesn't show any joystick output. 
It only works when I cast the result to int32_t like this:
`#define JOYSTICK_RESOLUTION (int32_t)((1UL << (JOYSTICK_AXES_RESOLUTION - 1)) - 1)`

Reply:
That might be okay, but I'm not sure why the original doesn't work... 1 << 15 would put the 1 at the very top bit, and shouldn't overflow :/

Reply:
With `1U`  it works like a charm.
As to the original, the error message says `integer overflow in expression '-32768 - 1' of type 'int' results in '32767' [-Werror=overflow]`
So it seems it's indeed a 16 bit int, and then 1 << 15 is -32678 and subtracting 1 makes it overflow (underflow?) to 32676, which throws the error.

Reply:
Yeah, I guess it was just because integer literals are signed by default.